dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/RumEventData.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-2020 Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal.domain
+
+internal sealed class RumEventData(val category: String) {
+
+    internal data class Resource(
+        val kind: Kind,
+        val url: String,
+        val durationNanoSeconds: Long
+    ) : RumEventData("resource") {
+
+        internal enum class Kind(val value: String) {
+            UNKNOWN("unknown"),
+            XHR("xhr"),
+            FETCH("fetch"),
+            IMAGE("image"),
+            JS("js"),
+            FONT("font"),
+            CSS("css"),
+            BEACON("beacon"),
+            MEDIA("media"),
+            DOCUMENT("document"),
+            OTHER("other");
+        }
+    }
+
+    internal data class UserAction(
+        val name: String
```

Comment:
don't we need also a duration on the UserAction ?

Reply:
Yes but it'll be on a separate ticket, I'd like to get this one merged and build upon it